container.go

```diff
@@ -103,8 +105,17 @@ type ProcessListItem struct {
 	UserTime100ns                uint64    `json:",omitempty"`
 }
 
+// createContainerAdditionalJSON is read from the environment at initialisation
+// time. It allows an environment variable to define additional JSON which
+// is merged in the CreateContainer call to HCS.
+var createContainerAdditionalJSON string
+
+func init() {
+	createContainerAdditionalJSON = os.Getenv("HCSSHIM_CREATECONTAINER_ADDITIONALJSON")
+}
+
 // CreateContainer creates a new container with the given configuration but does not start it.
-func CreateContainer(id string, c *ContainerConfig) (Container, error) {
+func CreateContainer(id string, c *ContainerConfig, additionalJSON string) (Container, error) {
```

Comment:
To avoid a breaking change to the interface, maybe it's better to create a second function, CreateContainerWithJsonTemplate or something.